.github/workflows/ci.yml

```diff
@@ -237,7 +237,7 @@ jobs:
         run: cargo fmt --all --check
 
       - name: Run cargo clippy
-        run: cargo clippy --all-features --all-targets -- -D warnings -A incomplete-features
```

Comment:
Any issue with the `--all-features` clippy? I think it should stay, no harm done, and easier to spot things.

Reply:
We can't have 2 network features on `evm_arithmetization` because that means we want to support 2 networks at once.

For now our whole CI is running assuming Eth mainnet, but we soon will need to rework it to ensure consistency across all supported networks.

Reply:
(Knowing that very soon, once i'm done with #562, we will have the new network features also in upper crates, which will prevent to use `all-features` basically everywhere)

Reply:
> Removing --all-features means only default features would be tested with clippy, e.g. if you develop code for polygon-pos it would never be checked. 

Yes I know, this is what I was implying above. For now we support _only_ the default `eth_mainnet` network, so it does not hurt to remove the `--all-features` flag anyway, as we would need a whole new CI to handle testing on both `cdk` / `eth` / `polygon` chains.

> Alternative networks features in the Cargo.toml would be organized in that way that they are not conflicting, e.g. if turning on polygon-pos would not compile eth_mainnet and clippy would understand that.

This is already done in this PR within the `evm_arithmetization` crate at the root level, and will be replicated within the other crates with #562).

Reply:
Note that the current use of _only_ testing `clippy` against `--all-features` is far from perfect. It checks the zero-bin/leader` crate _only_ with `cdk_erigon` feature enabled because of this, but not the default setup that we always use.